tests/unit/cli/storage_layout/test_storage_layout_overrides_fuzzing.py

```diff
@@ -0,0 +1,254 @@
+
+    return type_definitions, declarations
+
+
+# sanity check a storage layout
+# check that variables don't overlap and that each var starts
+# where the previous one ends (doesn't account for hash ptrs)
+def validate_storage_layout(layout_dict):
+    for section in ["storage_layout", "transient_storage_layout"]:
+        if section not in layout_dict:
+            continue
+
+        variables = [info for _, info in layout_dict[section].items()]
+
+        counter = variables[0]["slot"]
+        for info in variables:
+            if info["slot"] != counter:
+                raise ValueError("Invalid layout")
+            counter += info["n_slots"]
+
+
+@st.composite
+def _select_section(draw, layout):
+    if not ENABLE_TRANSIENT:
+        return layout["storage_layout"]
+    section = draw(st.sampled_from(["storage_layout", "transient_storage_layout"]))
+    if section not in layout or not layout[section]:
+        section = "transient_storage_layout" if section == "storage_layout" else "storage_layout"
+    return layout[section]
+
+
+@st.composite
+def drop_random_item(draw, layout) -> MutationResult:
+    result = layout
+    section = draw(_select_section(layout))
+
+    if len(section) == 0:
+        return MutationResult(False, result)
+
+    keys = st.sampled_from(list(section.keys()))
+    item_name = draw(keys)
+    del section[item_name]
+
+    return MutationResult(success=True, layout=result)
+
+
+@st.composite
+def mutate_slot_address(draw, layout) -> MutationResult:
+    result = layout
+    section = draw(_select_section(layout))
+
+    if len(section) < 2:
+        return MutationResult(False, result)
+
+    items = list(section.keys())
+    item_to_change = draw(st.sampled_from(items[:-1]))
+    last_slot = section[items[-1]]["slot"]
+    assert last_slot > 0
+    strategy = st.integers(0, last_slot)
+    new_slot = draw(strategy)
+    while section[item_to_change]["slot"] == new_slot:
+        new_slot = draw(strategy)
+    section[item_to_change]["slot"] = new_slot
+
+    return MutationResult(success=True, layout=result)
+
+
+@st.composite
+def mutate_slot_size(draw, layout) -> MutationResult:
+    result = layout
+    section = draw(_select_section(layout))
+
+    if len(section) < 2:
+        return MutationResult(False, result)
+
+    items = list(section.keys())
+    item_to_change = draw(st.sampled_from(items[:-1]))
+    last_slot = section[items[-1]]["slot"]
+    strategy = st.integers(-last_slot, last_slot)
+    delta = draw(strategy)
+    while delta == 0:
+        delta = draw(strategy)
+    section[item_to_change]["n_slots"] = section[item_to_change]["n_slots"] + delta
+
+    return MutationResult(success=True, layout=result)
+
+
+@st.composite
+def mutate_layout(draw, layout):
+    mutation_strategies = [drop_random_item, mutate_slot_address, mutate_slot_size]
+
+    mutation = draw(st.sampled_from(mutation_strategies))
+
+    result = layout
+
+    mutation_result = draw(mutation(result))
```

Comment:
my intuition is that drawing from a strategy which is itself the result of a draw is going to cause performance problems. maybe represent the mutation strategies with characters like `"d", "ma", "ms"` and then switch on them?